### ISSUE TITLE
asm3.conf example is not exhaustive

### ISSUE THREAD

**Issue by inghamn** [bobintetley/asm3#581]
This may only be useful for Ansible work, however, it might be useful for people doing manual deployments, as well.

A challenge I have when writing Ansible deployments that work for everyone, is that I keep leaving out various asm3.conf variables.  I originally worked from asm3.conf.example; however, that file does not contain examples for all possible values.

It would be nice if, during the dist step, an exhaustive asm3.conf.example was prepared.  I'm thinking that one possible easy wasy to do this would be to extract all the key values and their default values from sitedefs.py.

A naive implementation could be to run through file with a regex parser.
* Comment lines would get added to an asm3.conf.example as is.
* The lines assigning the constant values could have their keys and default values extracted with the regex.
* Whitespace lines would be preserved.

The end result should look a lot like the current asm3.conf.example.  I think that one benefit of this approach is it creates a single file to add new configurations and comments about them.  (As opposed to trying to remember to update the example conf with new configurations as they get added.)

**Comment by inghamn** [comment]
I'm thinking of something along the lines of:

```bash
awk '/^#|^\s*$|^[A-Z_]+ = get_.+$/' sitedefs.py | sed -E 's/^([A-Z_]+) = get_[a-z]+\(\"(.+)\"\, [\x22\x27]?(.+)[\x22\x27]?\)/\2 = \3/'
```

This doesn't exactly work, as it doesn't handle all the possible ways values can be quoted, or not.

**Comment by bobintetley** [comment]
In my opinion, this seems overly complex and error prone. A simpler fix might be a simple comment/reminder in sitedefs.py that if you add something, you update asm3.conf.example

**Comment by inghamn** [comment]
You're right, this doesn't make sense for general purpose installations.  I'll just do something to generate a templated asm3.conf for Ansible to use.

**Comment by inghamn** [comment]
The templated Ansible version of asm3.conf has been working fine
